File: 프로그래머스/2/42578. 의상/의상.py
import itertools

def solution(clothes):
    dic = {}
    
    for cloth in clothes:
        key = cloth[1]
        value = cloth[0]
        if not key in dic:
            dic[key] = []
        
        dic[key].append(value)
        
        
    print(dic)
        
    value_list = list(dic.values())

    
    a = itertools.product(*value_list)

    

    return len(list((a)))

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String[] cloth: clothes) {
            String key = cloth[1];
            map.put(key, map.getOrDefault(key, 0) + 1);
        }
        
        
        int answer = 1;
        
        for (int x: map.values()) {
            answer *= x + 1;
        }
        
    
        return answer - 1;
    }
}